src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -516,7 +515,16 @@ private void SendKeepAliveFrameAsync()
             {
                 // This exists purely to keep the connection alive; don't wait for the result, and ignore any failures.
                 // The call will handle releasing the lock.
-                SendFrameLockAcquiredNonCancelableAsync(MessageOpcode.Ping, true, new ArraySegment<byte>(Array.Empty<byte>()));
+                Task t = SendFrameLockAcquiredNonCancelableAsync(MessageOpcode.Ping, true, new ArraySegment<byte>(Array.Empty<byte>()));
+
+                // "Observe" any exception, ignoring it to prevent the unobserved exception event from being raised.
+                if (!t.IsCompletedSuccessfully)
+                {
+                    t.ContinueWith(p => { Exception ignored = p.Exception; },
```

Comment:
It would be good to log such exceptions with NetEventSource.

Reply:
Good suggestion.  Will do.

Reply:
Ah, we don't currently use NetEventSource in this project.  I'll open an issue to do so, and this can be addressed then.